airflow/triggers/temporal.py

```diff
@@ -68,9 +68,8 @@ async def run(self):
         while self.moment > timezone.utcnow():
             self.log.info("sleeping 1 second...")
             await asyncio.sleep(1)
-        # Send our single event and then we're done
-        self.log.info("yielding event with payload %r", self.moment)
-        yield TriggerEvent(self.moment)
+        self.log.info("Sensor time condition reached; marking task successful and exiting")
+        yield TaskSuccessEvent()
```

Comment:
here we have implemented for timedelta sensor

Reply:
ok so... there's a potential backcompat problem here

i think it's pretty unlikely but you never know

case is, user subclasses this operator and does some actual work in execute complete...  

do you think we need to worry about that @potiuk @uranusjr

Reply:
to me i think that is venturing into xkcd 1172 territory

Reply:
actually yeah... i think we discussed this a while back and agreed (though may not have gotten around to documenting)

our backcompat guarantee for operators is about their behavior, not the exact structure of all of their methods. we don't publish them as "frameworks" to build upon -- they are end products.

so i think the operators are not a concern, but potentially the trigger class itself could be a concern since it now exits the task instead of sending back to task.  so at most, if we need to do anything, we would need either to create another version of DateTimeTrigger which exits, or make it parameterizable.  but we would be free to update the opreators to use this new behavior since it does not affect overall behavior, even if it leaves a legacy method unused (execute_complete, that is).

Reply:
ok so ... i updated to make DateTimeTrigger be backward compatible. you use `exit_task=True` if you want it to exit instead of send back to task.